pkg/registry/servicecatalog/instance/strategy.go

```diff
@@ -181,3 +198,23 @@ func (instanceStatusRESTStrategy) ValidateUpdate(ctx genericapirequest.Context,
 
 	return scv.ValidateServiceInstanceStatusUpdate(newServiceInstance, oldServiceInstance)
 }
+
+// setServiceInstanceUserInfo injects user.Info from the request context
+func setServiceInstanceUserInfo(instance *sc.ServiceInstance, ctx genericapirequest.Context) {
+	instance.Spec.UserInfo = nil
+	if utilfeature.DefaultFeatureGate.Enabled(scfeatures.OriginatingIdentity) {
```

Comment:
I would rather have feature gate checked outside this method at the call site, that way it's clear at the call site that the gate has to be enabled.